A66-otel-stats.md

```diff
@@ -0,0 +1,305 @@
+*   **grpc.server.call.rcvd_total_compressed_message_size** <br>
+    Total bytes (compressed but not encrypted) received across all request messages (metadata excluded) per RPC; does not include grpc or transport framing bytes. <br>
+    *Attributes*: grpc.method, grpc.authority, grpc.status <br>
+    *Type*: Histogram (Size Buckets) <br>
+*   **grpc.server.call.duration** <br>
+    This metric aims to measure the end2end time an RPC takes from the server transport’s (HTTP2/ inproc / cronet) perspective. <br>
+    Start timestamp - After the transport knows that it's got a new stream. For HTTP2, this would be after the first header frame for the stream has been received and decoded. Whether the timestamp is recorded before or after HPACK is left to the implementation. <br>
+    End timestamp - Ends at the first point where the transport considers the stream done. For HTTP2, this would be when scheduling a trailing header with END_STREAM to be written, or RST_STREAM, or a connection abort. Note that this wouldn’t necessarily mean that the bytes have also been immediately scheduled to be written by TCP. <br>
+    *Attributes*: grpc.method, grpc.authority, grpc.status <br>
+    *Type*: Histogram (Latency Buckets) <br>
+
+## Language Specifics
+
+Each language implementation will provide an API for registering an
+OpenTelemetry plugin. Overall, the APIs should have the following capabilities -
+
+*   Allow installing multiple OpenTelemetry plugins.
+*   Allow setting a
+    [MeterProvider](https://opentelemetry.io/docs/specs/otel/metrics/api/#meterprovider)
+    on individual plugins.
+*   Optionally allow enabling/disabling metrics. This would allow optimizations
+    to avoid computation and collection of expensive stats within the gRPC
+    library. Note that even without this capability, users of OpenTelemetry
+    would be able to customize
+    [views](https://opentelemetry.io/docs/specs/otel/metrics/sdk/#view) through
+    the MeterProvider.
+*   Optionally allow setting of a OpenTelemetry plugin for a specific channel or
+    server, instead of setting it globally.
+
+Note that implementations of the gRPC OpenTelemetry plugin
+[should prefer](https://opentelemetry.io/docs/specs/otel/overview/) to only
+depend on the OpenTelemetry API and not the OpenTelemetry SDK.
+
+### C++
+
+```c++
+Class OpenTelemetryPluginBuilder {
+ public:
+  // Enables base set of metrics by default
+  OpenTelemetryPluginBuilder() = default;
+  // If `SetMeterProvider` is called, \a meter_provider will be used instead of the global default.
+  OpenTelemetryPluginBuilder& SetMeterProvider(shared_ptr<opentelemetry::metrics::MeterProvider> meter_provider);
+  // Enable metric \a metric_name.
+  OpenTelemetryPluginBuilder&  EnableMetric(absl::string_view metric_name);
+  // Disable metric \a metric_name
+  OpenTelemetryPluginBuilder&  DisableMetric(absl::string_view metric_name);
+  // Builds and registers a OpenTelemetry Plugin
+  void BuildAndRegisterGlobal();
```

Comment:
Since this is a builder pattern why not have a `build` return an instance and then a separate `register`? Also should e provider a "shutdown" hook for a plugin?

Reply:
Yeah, this is still under flux. Suggestions welcome